### ISSUE TITLE
Error in using --format=wpapsk-opencl in john/kernels/opencl_cmac.h

### ISSUE THREAD

**Issue by XRFXLP** [openwall/john#5196]
Related to https://github.com/openwall/john/issues/4667, but that seems to be fixed

Problem: 
```bash
$ john --format=wpapsk-opencl --test                                                                                                                              
Device 1@box: NVIDIA GeForce RTX 3050 Laptop GPU
Benchmarking: wpapsk-opencl, WPA/WPA2/PMF/PMKID PSK [PBKDF2-SHA1 OpenCL]... Options used: -I /usr/share/john/kernels -cl-mad-enable -DSM_MAJOR=8 -DSM_MINOR=6 -D__GPU__ -DDEVICE_INFO=18 -D__SIZEOF_HOST_SIZE_T__=8 -DDEV_VER_MAJOR=515 -DDEV_VER_MINOR=65 -D_OPENCL_COMPILER -DHASH_LOOPS=105 -DITERATIONS=4095 -DPLAINTEXT_LENGTH=63 -DV_WIDTH=1 /usr/share/john/kernels/wpapsk_kernel.cl
Build log: In file included from <kernel>:13:
/usr/share/john/kernels/opencl_cmac.h:62:22: error: implicit conversion from address space "generic" to address space "private" is not supported when passing to parameter of destination type 
        AES_set_encrypt_key(key, 128, &ctx->aesctx);
                            ^~~
/usr/share/john/kernels/opencl_aes_bitslice.h:1001:61: note: expanded from macro 'AES_set_encrypt_key'
#define AES_set_encrypt_key(key, bits, ctx) AES_Setkey(ctx, key, (bits) / 8)
                                                            ^
/usr/share/john/kernels/opencl_aes_bitslice.h:690:45: note: passing argument to parameter 'key' here
AES_Setkey(AES_CTX *ctx, AES_KEY_TYPE void *key, int len)
                                            ^

Error building kernel /usr/share/john/kernels/wpapsk_kernel.cl. DEVICE_INFO=18
0: OpenCL CL_BUILD_PROGRAM_FAILURE (-11) error in opencl_common.c:1386 - clBuildProgram
```

Configuration: 
```zsh
$ pacman -Q | grep nvidia                                                                                                                                          
lib32-nvidia-utils 515.65.01-1
linux419-nvidia 515.65.01-5
linux515-nvidia 515.65.01-8
linux519-nvidia 515.65.01-13
mhwd-nvidia 515.65.01-3
mhwd-nvidia-390xx 390.154-1
mhwd-nvidia-470xx 470.141.03-1
nvidia-prime 1.0-4
nvidia-utils 515.65.01-3
opencl-nvidia 515.65.01-3

$ john --list=build-info                                                                                                                                             ✔ 
Version: 1.9.0-jumbo-1
Build: linux-gnu 64-bit x86_64 AVX AC MPI + OMP
SIMD: AVX, interleaving: MD4:3 MD5:3 SHA1:1 SHA256:1 SHA512:1
System-wide exec: /usr/lib/john
System-wide home: /usr/share/john
Private home: ~/.john
CPU tests: AVX
CPU fallback binary: john-non-avx
$JOHN is /usr/share/john/
Format interface version: 14
Max. number of reported tunable costs: 4
Rec file version: REC4
Charset file version: CHR3
CHARSET_MIN: 1 (0x01)
CHARSET_MAX: 255 (0xff)
CHARSET_LENGTH: 24
SALT_HASH_SIZE: 1048576
SINGLE_IDX_MAX: 2147483648
SINGLE_BUF_MAX: 4294967295
Effective limit: Number of salts vs. SingleMaxBufferSize
Max. Markov mode level: 400
Max. Markov mode password length: 30
gcc version: 12.1.0
GNU libc version: 2.35 (loaded: 2.36)
OpenCL headers version: 1.2
Crypto library: OpenSSL
OpenSSL library version: 0101010ff      (loaded: 01010111f)
OpenSSL 1.1.1o  3 May 2022      (loaded: OpenSSL 1.1.1q  5 Jul 2022)
GMP library version: 6.2.1
File locking: fcntl()
fseek(): fseek
ftell(): ftell
fopen(): fopen
memmem(): System's

$ uname -r                                                                                                                                                           ✔ 
5.19.7-1-MANJARO

```

**Comment by ghost** [comment]
Thanks for reporting the issue.

> Related to #4667, but that seems to be fixed

You are using an old version of `john`, the Jumbo 1, it was released before #4667 was fixed.

Could you please try to use the latest code from this github repository? Could you build from the sources or use a newer package?

Please test and report back.

**Comment by XRFXLP** [comment]
Thanks!
